File: isAnagram.py
#Hashmap/Dictionary solution
#Time complexity -> O(n) uses forloop
#Space complextiy -> O(n) because uses hashmaps
class Solution:
    def isAnagram(self, s: str, t: str):
        if len(s) != len(t):
            return False
        dictionaryS = {}
        dictionaryT = {}
        for i in range(len(s)):
            dictionaryS[s[i]] = 1 + dictionaryS.get(s[i], 0)
            dictionaryT[t[i]] = 1 + dictionaryT.get(t[i], 0)
        
        return dictionaryS == dictionaryT

##################################################################
##################################################################

#Sorted function
#Time complexity -> O(nlogn) becasue we are using the sorted algoritm
#Space complexity -> O(1) because we aren't using any data structure
class Solution:
    def isAnagram(self, s: str, t: str):
        if len(s) != len(t):
            return False
        
        return sorted(s) == sorted(t)

File: isAnagram.cpp
// Hashmap
//Time Complexity -> O(n) uses a for loop
//Space Complexity -> O(n) uses an unordered_map (hashmap)

using namespace std;

class Solution {
    bool isAnagram(string s, string t){
        if (s.length() != t.length()){
            return false;
        }

        unordered_map<char, int> mapS;
        unordered_map<char, int> mapT;

        for(int i = 0; i < s.length(); i++){
            mapS[s[i]]++;
            mapT[t[i]]++;
        }

        return mapS == mapT;
    }
}

/////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////

//Sorted way
//Time Complexity -> O(nlogn) because we are using the sort algorithm
//Space Complexity -> O(1)

#include <algorithm>
class Solution{
    bool isAnagram(string s, string t){
        if (s.length() != t.length()){
            return false;
        }

        return sort(s.begin(), s.end()) == sort(t.begin(), t.end())
    }
}